crates/workers/src/workers.rs

```diff
@@ -106,20 +145,28 @@ impl Workers {
                 let worker_id = key_pair.get_peer_id();
 
                 let worker_info = self
-                    .store_worker(worker_id, deal_id.clone(), init_peer_id)
+                    .store_worker(worker_id, deal_id.clone(), init_peer_id, cu_count)
                     .await;
 
                 match worker_info {
                     Ok(worker_info) => {
                         let mut worker_ids = self.worker_ids.write();
                         let mut worker_infos = self.worker_infos.write();
+                        let mut runtimes = self.runtimes.write();
 
                         if worker_ids.contains_key(&deal_id) {
                             return Err(WorkersError::WorkerAlreadyExists { deal_id });
                         }
 
+                        let runtime = tokio::runtime::Builder::new_multi_thread()
+                            .worker_threads(params.cu_count)
+                            .max_blocking_threads(params.cu_count)
+                            .build()
+                            .map_err(|err| WorkersError::CreateRuntime { err })?;
+
```

Comment:
Why take locks before checking the error and creating a runtime?

Maybe like this instead? The suggestion simply moves lines 154, 155 to 166, 167.
```suggestion
                        if worker_ids.contains_key(&deal_id) {
                            return Err(WorkersError::WorkerAlreadyExists { deal_id });
                        }

                        let runtime = tokio::runtime::Builder::new_multi_thread()
                            .worker_threads(params.cu_count)
                            .max_blocking_threads(params.cu_count)
                            .build()
                            .map_err(|err| WorkersError::CreateRuntime { err })?;
                            
                        let mut worker_infos = self.worker_infos.write();
                        let mut runtimes = self.runtimes.write();
```

Reply:
Fixed